### ISSUE TITLE
Default os seems to be broken if conan is invoked from a cygwin shell

### ISSUE THREAD

**Issue by berncz** [conan-io/conan#1426]
I am trying to invoke conan install from a Cygwin shell within a virtualenv to install the "cmaker_installer" packet (cmake_installer/0.1@lasote/testing). Unfortunately, the installation process fails with the following error:

"ERROR: Invalid setting 'CYGWIN_NT-6.1-WOW' is not a valid 'settings.os' value.
Possible values are ['Android', 'FreeBSD', 'Linux', 'Macos', 'SunOS', 'Windows', 'iOS']
Read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-invalid-setting"

Since also conan is installed in a virtualenv (pip install), a new "conan.conf" file gets generated as soon as I invoke conan for the first time. This file already contains the invalid OS setting. As a workaround, I have a tried to overwrite the os setting ("conan install -s os=Windows" ), but this seems to have no impact at all.

**Comment by memsharded** [comment]
Can you please try to directly modify the "os" in [default_settings] in ``conan.conf`` file? That would be useful to know if it is only a detection error (automatic detection is only necessary to generate that file, after that, it is not used at all). Please tell me, thanks!

**Comment by berncz** [comment]
If I change the default OS, I get the following error:
ERROR: Conanfile: 'settings.compiler' value not defined

**Comment by lasote** [comment]
And do you have some detected compiler in your `conan.conf` file?

**Comment by memsharded** [comment]
You have to have some compiler defined. Either as the default or in the command line.
However, if you are just installing the mingw package, you might be able to do it without having a compiler defined, because the mingw package does not have "compiler" as setting. 
You can try running ``conan install mingw_installer/0.1@lasote/testing -g virtualrunenv``. 

I guess you are installing more packages, or installing from a conanfile.txt or something like that. Most likely the detect step failed to identify your compiler (gcc should be in the path, and callable).

**Comment by nesono** [comment]
I faced the same issue as berndcz. The compiler was gcc (as found in Cygwin) with the corresponding settings (I haven't verified them though). Manually setting the contents of conan.config to correct values made the call succeed. I assume the auto detection should not result in invalid settings though.

**Comment by memsharded** [comment]
Hi @nesono, @berncz , yes I agree, the auto detection shouldn't result in invalid settings. But the truth is that with the large variety of systems, compilers, etc., it is almost impossible to achieve that without the users help. As solving it is so easy, it requires just a few seconds to fix in the conan.conf file, it doesn't make sense that we spend lots of time trying to setup any configuration we can think of, but rather work on more critical issues.

So, please, it would be great if you could help improving this. Running conan from source is not difficult (https://github.com/conan-io/conan#from-source), and to test it you could do:
- Remove manually the file conan.conf
- Run ``conan``. 
The source code is in this file: https://github.com/conan-io/conan/blob/develop/conans/client/detect.py

Please tell if you need any help. Thanks!

**Comment by nesono** [comment]
Added Pull Request #1474 that addresses this issue.

**Comment by SSE4** [comment]
I think this one is already addressed, as for conan 1.9.2. executing `conan profile new 1 --detect && conan profile show 1` within Cygwin shell results in:
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]
```
both os and compiler are correctly detected

**Comment by lasote** [comment]
Thanks @SSE4 I'm closing this issue, please @berncz if you think this is not fixed comment and we will reopen it.
